src/containers/permissions/ObjectPermissionsCardStack.js

```diff
@@ -0,0 +1,93 @@
+/*
+ * @flow
+ */
+
+import React, { useMemo } from 'react';
+
+import { List, Map } from 'immutable';
+import { Types } from 'lattice';
+import { Typography } from 'lattice-ui-kit';
+import { useSelector } from 'react-redux';
+import type {
+  Ace,
+  Organization,
+  PermissionType,
+  Principal,
+  Role,
+  UUID,
+} from 'lattice';
+
+import ObjectPermissionsCard from './ObjectPermissionsCard';
+
+import { selectOrganization, selectOrganizationMembers } from '../../core/redux/selectors';
+import { getPrincipal } from '../../utils';
+
+const { PrincipalTypes } = Types;
+
+const ObjectPermissionsCardStack = ({
+  filterByPermissionTypes,
+  filterByQuery,
+  organizationId,
+  permissions,
+} :{
+  filterByPermissionTypes :Array<PermissionType>;
+  filterByQuery :string;
+  organizationId :UUID;
+  permissions :List<Ace>;
+}) => {
+
+  const organization :?Organization = useSelector(selectOrganization(organizationId));
+  const organizationRoles :Map<Principal, Role> = useMemo(() => (
+    Map().withMutations((mutableMap :Map<string, Role>) => {
+      organization?.roles.forEach((role :Role) => {
+        mutableMap.set(role.principal, role);
+      });
+    })
+  ), [organization]);
+
+  const members :List<Map> = useSelector(selectOrganizationMembers(organizationId));
+  const organizationMembers :Map<Principal, Map> = useMemo(() => (
+    Map().withMutations((mutableMap :Map<string, Map>) => {
+      members.forEach((member :Map) => {
+        const principal :?Principal = getPrincipal(member);
+        if (principal) {
+          mutableMap.set(principal, member);
+        }
+      });
+    })
+  ), [members]);
```

Comment:
I'm curious how useMemo and other hooks compare equality between immutable collection.

Reply:
React uses `Object.is`:

https://reactjs.org/docs/hooks-reference.html#bailing-out-of-a-state-update
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/is#Description

Here, the `members` object should be the same across renders because it's not being "computed". It comes from the redux store, so we expect to get the same object every time.